util/reggen/countermeasure.py

```diff
@@ -126,3 +130,58 @@ def _asdict(self) -> Dict[str, object]:
     def __str__(self) -> str:
         namestr = self.asset + '.' + self.cm_type
         return self.instance + '.' + namestr if self.instance else namestr
+
+    @staticmethod
+    def from_rtl_file(what: str,
+                      infile: str) -> List['CounterMeasure']:
+        """
+        Extracts a list of countermeasures from RTL annotations
+        """
+        matches = re.findall(r'//\s' + CM_RTL_TAG + r'\s(.*)',
+                             infile, re.MULTILINE)
+        retlist = []
+        if matches:
+            for m in matches:
+                cm = CounterMeasure.from_raw(what,
+                                             {'name': m,
+                                              'desc': 'extracted from RTL'})
+                retlist.append(cm)
+        return retlist
+
+    @staticmethod
+    def check_annotation_list(what: str,
+                              annotation_list: List['CounterMeasure'],
+                              hjson_list: List['CounterMeasure']):
```

Comment:
This is much easier if we do the rejig I've suggested above:
```python
    @staticmethod
    def check_annotation_list(what: str,
                              rtl_names: Dict[str, List[Tuple[str, int]]],
                              hjson_list: List['CounterMeasure']) -> None:
        """Compare found list of countermeasures against list from Hjson

        This compares a dictionary of countermeasure names extracted from the
        RTL against the list defined in the IP Hjson and checks that they
        match: every name in the RTL should correspond to an entry in the Hjson
        and every entry in the Hjson should have at least one matching name in
        the RTL.

        """
        hjson_set = {str(cm) for cm in hjson_list}
        rtl_set = set(rtl_names.keys())

        # Is there anything in the RTL that doesn't correspond to an Hjson
        # entry?
        for name in rtl_set - hjson_set:
            # Print out an error message for each hit and then raise a
            # RuntimeError.
            for path, line in rtl_names[name]:
                log.error('Unknown countermeasure {!r} '
                          'referenced at {}, line {}.'
                          .format(name, path, line))
            raise RuntimeError('One or more unknown countermeasures '
                               'referenced in RTL.')

        # Is there anything in the Hjson that isn't described in the RTL?
        for name in hjson_set - rtl_set:
            log.warning("Countermeasure {} is referenced in the Hjson, but "
                        "doesn't appear in the RTL."
                        .format(name))

        # TODO: Once all designs are annotated, generate a RuntimeError if we
        #       saw anything in the loop above.
```

Reply:
The set subtraction is pretty neat! 
And the line annotation could come in very handy, also later on if we decide to augment our autogen'd tables with links to lines.